bottles: update version + bugfixes

asbachb opened the pull request:
###### Motivation for this change
Previous state of `bottles` was not able to run any `wine` related commands beside installation of a wine prefix.

As it seems `staging-next` is stuck for some time and I want to get rid of that changes I want to merge this PR into `staging`.

see #132342

###### Things done
<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
```
 asbachb@nixo-t14s  ~/dev/src/nix/nixpkgs   update/bottles  nix-shell -p nixpkgs-review --run "nixpkgs-review wip"
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0
$ git worktree add /home/asbachb/.cache/nixpkgs-review/rev-ecb3c84196017d3ea37c6e7ec9c56a82d237b719-dirty-1/nixpkgs 739f0b22b74a35f65e5b862108db4d4380feecc8
Preparing worktree (detached HEAD 739f0b22b74)
Updating files: 100% (27423/27423), done.
HEAD is now at 739f0b22b74 elixir_ls: 0.7.0 -> 0.8.0
$ nix-env --option system x86_64-linux -f /home/asbachb/.cache/nixpkgs-review/rev-ecb3c84196017d3ea37c6e7ec9c56a82d237b719-dirty-1/nixpkgs -qaP --xml --out-path --show-trace
No diff detected, stopping review...
$ git worktree prune
```
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

asbachb committed:
bottles: 2021.7.14-treviso -> 2021.7.28-treviso-2

asbachb committed:
bottles: Added `xdg-utils` to enable browsing wine prefix files

asbachb committed:
bottles: Add `pciutils` as `bottles` uses it to determine the graphics environment

asbachb committed:
bottles: Ensure `wine` is run inside `steam-run`

asbachb commented:
/marvin opt-in
/status needs_review

asbachb commented:
/status needs_reviewer

asbachb commented:
This was already reviewed in https://github.com/NixOS/nixpkgs/pull/132342

mschuwalow commented on pkgs/applications/misc/bottles/default.nix:
```diff
@@ -52,7 +52,12 @@ python3Packages.buildPythonApplication rec {
+  ] ++ [
+    steam-run-native
+    xdg-utils
+    pciutils
```
Played around with bottles a bit and seems like it also wants `cabextract` and `wineWowPackages.minimal`(or maybe more) for some operations.

`cabextract` is needed to install wants using the dependencies menu.
The wine packages are needed when switching dxvk or vkd3d in a bottle.

Would be good to add them here.

asbachb replied:
Yeah there might be additional packages missing. Before I picked up nothing worked at all. Just tested a small subset of functionality.
May I ask you to request a change. Since currently I've no real possibility to build that package based on staging.

mschuwalow commented on pkgs/applications/misc/bottles/default.nix:
```diff
@@ -3,18 +3,18 @@
 , python3Packages, gettext
 , appstream-glib, gdk-pixbuf, glib, gobject-introspection, gspell, gtk3
-, steam-run-native
+, steam-run-native, xdg-utils, pciutils
```
```suggestion
, steam-run, xdg-utils, pciutils, cabextract, winePackages
```

mschuwalow commented on pkgs/applications/misc/bottles/default.nix:
```diff
@@ -64,6 +69,8 @@ python3Packages.buildPythonApplication rec {
     substituteInPlace src/runner.py \
       --replace " {runner}" " ${steam-run-native}/bin/steam-run {runner}" \
```
```suggestion
      --replace " {runner}" " ${steam-run}/bin/steam-run {runner}" \
```

mschuwalow commented on pkgs/applications/misc/bottles/default.nix:
```diff
@@ -64,6 +69,8 @@ python3Packages.buildPythonApplication rec {
     substituteInPlace src/runner.py \
       --replace " {runner}" " ${steam-run-native}/bin/steam-run {runner}" \
       --replace " {dxvk_setup}" " ${steam-run-native}/bin/steam-run {dxvk_setup}"
```
```suggestion
      --replace " {dxvk_setup}" " ${steam-run}/bin/steam-run {dxvk_setup}"
```

mschuwalow commented on pkgs/applications/misc/bottles/default.nix:
```diff
@@ -64,6 +69,8 @@ python3Packages.buildPythonApplication rec {
     substituteInPlace src/runner.py \
       --replace " {runner}" " ${steam-run-native}/bin/steam-run {runner}" \
       --replace " {dxvk_setup}" " ${steam-run-native}/bin/steam-run {dxvk_setup}"
+      substituteInPlace src/runner_utilities.py \
+        --replace " {runner}" " ${steam-run-native}/bin/steam-run {runner}" \
```
```suggestion
        --replace " {runner}" " ${steam-run}/bin/steam-run {runner}" \
```

mschuwalow replied:
```suggestion
      --replace " {runner}" " ${steam-run}/bin/steam-run {runner}" \
      --replace " {dxvk_setup}" " ${steam-run}/bin/steam-run {dxvk_setup}"
      substituteInPlace src/runner_utilities.py \
        --replace " {runner}" " ${steam-run}/bin/steam-run {runner}" \
```

mschuwalow commented on pkgs/applications/misc/bottles/default.nix:
```diff
@@ -52,7 +52,12 @@ python3Packages.buildPythonApplication rec {
-  ] ++ [ steam-run-native ];
+    patool
+  ] ++ [
+    steam-run-native
```
```suggestion
    steam-run
    cabextract
    winePackages.minimal
```

mschuwalow reviewed (changes requested):
These were enough for me to get all UI operations and a couple of games working

asbachb committed:
bottles: added `cabextract` and `wineWowPacakge.minimal`

> cabextract is needed to install wants using the dependencies menu.
The wine packages are needed when switching dxvk or vkd3d in a bottle.

Based on review by mschuwalow

SuperSandro2000 commented:
@ofborg eval

asbachb commented:
@SuperSandro2000 Can you have a look? I guess evaluation was successful.
